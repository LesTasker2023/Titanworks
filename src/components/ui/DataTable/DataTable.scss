@import '../../../styles/design-tokens';

// DataTable Component Styles
// BEM Methodology: .dataTable__element--modifier
.dataTable {
  width: 100%;
  position: relative;
  // Variant styles
  &--default {
    border: var(--border-width-thin) solid var(--color-border);
  }
  &--success {
    border: var(--border-width-thin) solid var(--color-success-20);
    background-color: var(--color-success-5);
  }
  &--warning {
    border: var(--border-width-thin) solid var(--color-warning-20);
    background-color: var(--color-warning-5);
  }
  &--danger {
    border: var(--border-width-thin) solid var(--color-destructive-20);
    background-color: var(--color-destructive-5);
  }
  // Size variants
  &--sizeSm {
    font-size: var(--font-size-xs);
  }
  &--sizeDefault {
    font-size: var(--font-size-sm);
  }
  &--sizeLg {
    font-size: var(--font-size-base);
  }
  &--sizeXl {
    font-size: var(--font-size-lg);
  }
  // State styles
  &--disabled {
    opacity: 0.5;
    pointer-events: none;
  }
  &--loading {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: var(--spacing-8);
    min-height: 200px;
    .dataTable__loadingContent {
      display: flex;
      align-items: center;
      gap: var(--spacing-2);
    }
    .dataTable__spinner {
      height: var(--spacing-6);
      width: var(--spacing-6);
      animation: spin 1s linear infinite;
    }
    .dataTable__loadingText {
      color: var(--color-muted-foreground);
    }
    .dataTable__srOnly {
      position: absolute;
      width: 1px;
      height: 1px;
      padding: 0;
      margin: -1px;
      overflow: hidden;
      clip: rect(0, 0, 0, 0);
      white-space: nowrap;
      border: 0;
    }
  }
  &--error {
    .dataTable__errorContent {
      padding: var(--spacing-4);
      text-align: center;
    }
    .dataTable__errorTitle {
      color: var(--color-destructive);
      font-weight: var(--font-weight-medium);
      margin-bottom: var(--spacing-2);
    }
    .dataTable__errorMessage {
      font-size: var(--font-size-sm);
      color: var(--color-muted-foreground);
    }
  }
  &__filterBar {
    padding: var(--spacing-4);
    border-bottom: 1px solid var(--color-border);
    background-color: var(--color-muted-30);
  }
  &__searchContainer {
    position: relative;
    max-width: 20rem;
  }
  &__searchIcon {
    position: absolute;
    left: var(--spacing-3);
    top: 50%;
    height: var(--spacing-4);
    width: var(--spacing-4);
    transform: translateY(-50%);
    color: var(--color-muted-foreground);
  }
  &__searchInput {
    width: 100%;
    padding-left: var(--spacing-10);
    padding-right: var(--spacing-4);
    padding-top: var(--spacing-2);
    padding-bottom: var(--spacing-2);
    border: var(--border-width-thin) solid var(--color-border);
    border-radius: var(--radius-md);
    background-color: var(--color-background);
    &:focus {
      outline: none;
      box-shadow: 0 0 0 2px var(--color-ring);
    }
    &:disabled {
      cursor: not-allowed;
      opacity: 0.5;
    }
  }
  &__scrollContainer {
    overflow: auto;
    scrollbar-width: thin;
    scrollbar-color: var(--color-muted) transparent;
    &::-webkit-scrollbar {
      height: 8px;
      width: 8px;
    }
    &::-webkit-scrollbar-track {
      background: var(--color-muted-10);
      border-radius: var(--radius);
    }
    &::-webkit-scrollbar-thumb {
      background-color: var(--color-muted);
      border-radius: var(--radius);
      &:hover {
        background-color: var(--color-muted-foreground);
      }
    }
  }
  &__table {
    width: 100%;
    caption-side: bottom;
    font-size: var(--font-size-sm);
  }
  &__caption {
    margin-top: var(--spacing-4);
    font-size: var(--font-size-sm);
    color: var(--color-muted-foreground);
  }
  &__thead {
    tr {
      border-bottom: 1px solid var(--color-border);
    }
    &--sticky {
      position: sticky;
      top: 0;
      background-color: var(--color-background);
      z-index: 10;
    }
  }
  &__th {
    height: var(--spacing-10);
    padding: 0 var(--spacing-2);
    text-align: left;
    vertical-align: middle;
    font-weight: var(--font-weight-medium);
    color: var(--color-muted-foreground);
    &--sortable {
      cursor: pointer;
      user-select: none;
      &:hover {
        background-color: var(--color-muted-50);
      }
    }
    &--sorted {
      background-color: var(--color-muted-30);
    }
    &--checkbox {
      width: var(--spacing-12);
    }
  }
  &__thContent {
    display: flex;
    align-items: center;
    gap: var(--spacing-1);
  }
  &__sortIndicator {
    display: flex;
    flex-direction: column;
    &--ascending .dataTable__chevronUp,
    &--descending .dataTable__chevronDown {
      color: var(--color-foreground);
    }
  }
  &__chevronUp,
  &__chevronDown {
    height: var(--spacing-3);
    width: var(--spacing-3);
    color: var(--color-muted-foreground);
  }
  &__chevronUp {
    margin-bottom: -0.25rem;
  }
  &__tbody {
    tr:last-child {
      border-bottom: none;
    }
  }
  &__row {
    border-bottom: 1px solid var(--color-border);
    transition: background-color 0.2s ease;
    &:hover {
      background-color: var(--color-muted-50);
    }
    &--selected {
      background-color: var(--color-accent-10);
      border-color: var(--color-ring);
    }
    &--clickable {
      cursor: pointer;
    }
    &--disabled {
      pointer-events: none;
    }
  }
  &__td {
    padding: var(--spacing-2);
    vertical-align: middle;
  }
  &__checkbox {
    border-radius: var(--radius);
    border: var(--border-width-thin) solid var(--color-border);
    &:disabled {
      cursor: not-allowed;
      opacity: 0.5;
    }
  }
  &__emptyState {
    padding: var(--spacing-8);
    text-align: center;
    color: var(--color-muted-foreground);
  }
  &__pagination {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: var(--spacing-4);
    border-top: 1px solid var(--color-border);
    background-color: var(--color-muted-30);
  }
  &__paginationInfo {
    font-size: var(--font-size-sm);
    color: var(--color-muted-foreground);
  }
  &__paginationControls {
    display: flex;
    align-items: center;
    gap: var(--spacing-2);
  }
  &__paginationButton {
    padding: var(--spacing-1) var(--spacing-3);
    border: var(--border-width-thin) solid var(--color-border);
    border-radius: var(--radius);
    background-color: var(--color-background);
    cursor: pointer;
    &:hover:not(:disabled) {
      background-color: var(--color-muted);
    }
    &:disabled {
      opacity: 0.5;
      cursor: not-allowed;
    }
  }
}
@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
