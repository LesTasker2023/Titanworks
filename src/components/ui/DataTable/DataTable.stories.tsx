import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import React from 'react';
import { DataTable } from './DataTable';
import './DataTable.scss';

// Sample data for stories
const sampleData = [
  { id: 1, name: 'John Doe', email: 'john@example.com', role: 'Admin', status: 'Active' },
  { id: 2, name: 'Jane Smith', email: 'jane@example.com', role: 'User', status: 'Active' },
  { id: 3, name: 'Bob Johnson', email: 'bob@example.com', role: 'User', status: 'Inactive' },
  { id: 4, name: 'Alice Brown', email: 'alice@example.com', role: 'Moderator', status: 'Active' },
  { id: 5, name: 'Charlie Wilson', email: 'charlie@example.com', role: 'User', status: 'Pending' },
];

const columns = [
  { key: 'name', header: 'Name', sortable: true },
  { key: 'email', header: 'Email', sortable: true },
  { key: 'role', header: 'Role', sortable: true },
  { key: 'status', header: 'Status', sortable: true },
];

const meta: Meta<typeof DataTable> = {
  title: 'UI/DataTable',
  component: DataTable,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: 'select',
      options: ['default', 'success', 'warning', 'danger'],
    },
    size: {
      control: 'select',
      options: ['sm', 'default', 'lg', 'xl'],
    },
    loading: { control: 'boolean' },
    disabled: { control: 'boolean' },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

// Required story types per style guide
export const Default: Story = {
  args: {
    data: sampleData,
    columns: columns,
  },
};

export const AllVariants: Story = {
  render: () => (
    <div className="space-y-8 w-full">
      <div>
        <h3 className="font-medium mb-2">Default Variant</h3>
        <DataTable variant="default" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
      <div>
        <h3 className="font-medium mb-2">Success Variant</h3>
        <DataTable variant="success" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
      <div>
        <h3 className="font-medium mb-2">Warning Variant</h3>
        <DataTable variant="warning" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
      <div>
        <h3 className="font-medium mb-2">Danger Variant</h3>
        <DataTable variant="danger" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
    </div>
  ),
};

export const AllSizes: Story = {
  render: () => (
    <div className="space-y-8 w-full">
      <div>
        <h3 className="font-medium mb-2">Small Size</h3>
        <DataTable size="sm" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
      <div>
        <h3 className="font-medium mb-2">Default Size</h3>
        <DataTable size="default" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
      <div>
        <h3 className="font-medium mb-2">Large Size</h3>
        <DataTable size="lg" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
      <div>
        <h3 className="font-medium mb-2">Extra Large Size</h3>
        <DataTable size="xl" data={sampleData.slice(0, 3)} columns={columns} />
      </div>
    </div>
  ),
};

export const LoadingState: Story = {
  args: {
    data: sampleData,
    columns: columns,
    loading: true,
  },
};

export const DisabledState: Story = {
  args: {
    data: sampleData,
    columns: columns,
    disabled: true,
  },
};

export const ErrorState: Story = {
  args: {
    data: sampleData,
    columns: columns,
    error: 'Failed to load data from server',
  },
};

export const InteractiveExample: Story = {
  render: () => {
    const [sortColumn, setSortColumn] = React.useState<string>('');
    const [sortDirection, setSortDirection] = React.useState<'asc' | 'desc'>('asc');

    return (
      <div className="w-full">
        <div className="mb-4 p-4 border rounded-lg bg-muted/30">
          <h3 className="font-medium mb-2">Interactive DataTable</h3>
          <p className="text-sm text-muted-foreground">
            Click column headers to sort. Current sort: {sortColumn || 'None'} ({sortDirection})
          </p>
        </div>
        <DataTable
          data={sampleData}
          columns={columns}
          sortable
          onSort={(column: string, direction: 'asc' | 'desc') => {
            setSortColumn(column);
            setSortDirection(direction);
          }}
          currentSort={{ column: sortColumn, direction: sortDirection }}
        />
      </div>
    );
  },
};

export const AccessibilityDemo: Story = {
  render: () => (
    <div className="w-full space-y-4">
      <div className="p-4 border rounded-lg bg-blue-50/50 dark:bg-blue-950/20">
        <h3 className="font-medium mb-2">♿ Accessibility Features</h3>
        <ul className="text-sm space-y-1 text-muted-foreground">
          <li>• Keyboard navigation with Tab/Arrow keys</li>
          <li>• Screen reader announcements for sorting</li>
          <li>• ARIA table structure with proper roles</li>
          <li>• Focus management and high contrast support</li>
        </ul>
      </div>
      <DataTable
        data={sampleData}
        columns={columns}
        sortable
        aria-label="User management table"
        caption="List of users with their roles and status"
      />
    </div>
  ),
};

// Additional enterprise stories
export const WithPagination: Story = {
  args: {
    data: Array.from({ length: 50 }, (_, i) => ({
      id: i + 1,
      name: `User ${i + 1}`,
      email: `user${i + 1}@example.com`,
      role: ['Admin', 'User', 'Moderator'][i % 3],
      status: ['Active', 'Inactive', 'Pending'][i % 3],
    })),
    columns: columns,
    pagination: true,
    pageSize: 10,
  },
};

export const WithSelection: Story = {
  args: {
    data: sampleData,
    columns: columns,
    selectable: true,
    onSelectionChange: (selectedIds: (number | string)[]) => {
      // In real app, handle selection change
      // setSelectedRows(selectedIds);
      if (process.env.NODE_ENV === 'development') {
        console.log('Selected rows:', selectedIds);
      }
    },
  },
};

export const WithFiltering: Story = {
  args: {
    data: sampleData,
    columns: columns,
    filterable: true,
    searchPlaceholder: 'Search users...',
  },
};
