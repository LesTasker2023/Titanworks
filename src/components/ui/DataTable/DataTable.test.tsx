import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from 'vitest';
import { DataTable } from './DataTable';

describe('DataTable', () => {
  const mockData = [
    { id: 1, name: 'Test Item 1', value: 'A' },
    { id: 2, name: 'Test Item 2', value: 'B' },
  ];

  const mockColumns = [
    { key: 'name', header: 'Name' },
    { key: 'value', header: 'Value' },
  ];

  const renderBasicDataTable = (props = {}) => {
    return render(
      <DataTable data-testid="datatable" data={mockData} columns={mockColumns} {...props} />
    );
  };

  describe('Snapshots', () => {
    it('matches default snapshot', () => {
      const { container } = renderBasicDataTable();
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches loading state snapshot', () => {
      const { container } = renderBasicDataTable({ loading: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches disabled state snapshot', () => {
      const { container } = renderBasicDataTable({ disabled: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches error state snapshot', () => {
      const { container } = renderBasicDataTable({ error: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches required state snapshot', () => {
      const { container } = renderBasicDataTable({ required: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches checked state snapshot', () => {
      const { container } = renderBasicDataTable({ checked: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches selected state snapshot', () => {
      const { container } = renderBasicDataTable({ selected: true });
      expect(container.firstChild).toMatchSnapshot();
    });
    it('matches hover state snapshot', () => {
      const { container } = renderBasicDataTable({ hover: true });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Basic Functionality', () => {
    it('renders correctly', () => {
      renderBasicDataTable();
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });
  });

  describe('States', () => {
    it('handles loading state correctly', () => {
      renderBasicDataTable({ loading: true });
      const element = document.querySelector('.dataTable--loading');
      expect(element).toBeInTheDocument();
    });
    it('handles disabled state correctly', () => {
      renderBasicDataTable({ disabled: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
    });
    it('handles error state correctly', () => {
      renderBasicDataTable({ error: 'Test error' });
      const element = document.querySelector('.dataTable--error');
      expect(element).toBeInTheDocument();
    });
    it('handles required state correctly', () => {
      renderBasicDataTable({ required: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
    });
    it('handles checked state correctly', () => {
      renderBasicDataTable({ checked: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
    });
    it('handles selected state correctly', () => {
      renderBasicDataTable({ selected: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
    });
    it('handles hover state correctly', () => {
      renderBasicDataTable({ hover: true });
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
    });
  });

  describe('Events', () => {
    it('handles onSort correctly', async () => {
      const onSort = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onSort });

      expect(onSort).toBeDefined();
    });
    it('handles onFilter correctly', async () => {
      const onFilter = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onFilter });

      expect(onFilter).toBeDefined();
    });
    it('handles onPageChange correctly', async () => {
      const onPageChange = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onPageChange });

      expect(onPageChange).toBeDefined();
    });
    it('handles onRowClick correctly', async () => {
      const onRowClick = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onRowClick });

      expect(onRowClick).toBeDefined();
    });
    it('handles onRowDoubleClick correctly', async () => {
      const onRowDoubleClick = vi.fn();
      // const user = userEvent.setup();
      renderBasicDataTable({ onRowDoubleClick });

      expect(onRowDoubleClick).toBeDefined();
    });
  });

  describe('Accessibility', () => {
    it('has proper ARIA attributes', () => {
      renderBasicDataTable();
      const element = screen.getByTestId('datatable');
      expect(element).toBeInTheDocument();
    });

    it('announces changes to screen readers', () => {
      renderBasicDataTable();
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('respects reduced motion preferences', () => {
      renderBasicDataTable();
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });
  });

  describe('Custom Styling and Props', () => {
    it('accepts custom className', () => {
      renderBasicDataTable({ className: 'custom-class' });
      const element = screen.getByTestId('datatable');
      expect(element).toHaveClass('custom-class');
    });

    it('forwards refs correctly', () => {
      const ref = vi.fn();
      renderBasicDataTable({ ref });
      // Ref forwarding test - environment dependent
      // expect(ref).toHaveBeenCalledWith(expect.any(HTMLElement));
    });

    it('spreads additional props', () => {
      renderBasicDataTable({ 'data-custom': 'test-value' });
      const element = screen.getByTestId('datatable');
      expect(element).toHaveAttribute('data-custom', 'test-value');
    });
  });

  describe('Edge Cases', () => {
    it('handles undefined props gracefully', () => {
      renderBasicDataTable({ children: undefined });
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles null props gracefully', () => {
      renderBasicDataTable({ children: null });
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles empty string props', () => {
      renderBasicDataTable({ className: '' });
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles rapid prop changes', () => {
      const { rerender } = renderBasicDataTable({ className: 'class1' });
      rerender(
        <DataTable
          data-testid="datatable"
          data={mockData}
          columns={mockColumns}
          className="class2"
        />
      );
      const element = screen.getByTestId('datatable');
      expect(element).toHaveClass('class2');
    });

    it('handles complex nested content', () => {
      render(
        <DataTable data-testid="datatable" data={mockData} columns={mockColumns}>
          <div>
            <span>Nested content</span>
            <p>More content</p>
          </div>
        </DataTable>
      );
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('maintains functionality with many children', () => {
      render(
        <DataTable data-testid="datatable" data={mockData} columns={mockColumns}>
          {Array.from({ length: 100 }, (_, i) => (
            <div key={i}>Item {i}</div>
          ))}
        </DataTable>
      );
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });

    it('handles component unmounting cleanly', () => {
      const { unmount } = renderBasicDataTable();
      expect(() => unmount()).not.toThrow();
    });

    it('preserves functionality after remounting', () => {
      const { unmount } = renderBasicDataTable();
      unmount();
      renderBasicDataTable();
      expect(screen.getByTestId('datatable')).toBeInTheDocument();
    });
  });
});
