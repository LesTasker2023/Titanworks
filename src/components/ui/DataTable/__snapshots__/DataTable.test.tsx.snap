// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTable > Snapshots > matches checked state snapshot 1`] = `
<div
  aria-disabled="false"
  class="dataTable dataTable--default dataTable--sizeDefault"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="dataTable__scrollContainer"
  >
    <table
      class="dataTable__table"
      role="table"
    >
      <thead
        class="dataTable__thead"
      >
        <tr>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="dataTable__tbody"
      >
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches default snapshot 1`] = `
<div
  aria-disabled="false"
  class="dataTable dataTable--default dataTable--sizeDefault"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="dataTable__scrollContainer"
  >
    <table
      class="dataTable__table"
      role="table"
    >
      <thead
        class="dataTable__thead"
      >
        <tr>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="dataTable__tbody"
      >
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches disabled state snapshot 1`] = `
<div
  aria-disabled="true"
  class="dataTable dataTable--default dataTable--sizeDefault dataTable--disabled"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="dataTable__scrollContainer"
  >
    <table
      class="dataTable__table"
      role="table"
    >
      <thead
        class="dataTable__thead"
      >
        <tr>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="dataTable__tbody"
      >
        <tr
          class="dataTable__row dataTable__row--disabled"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="dataTable__row dataTable__row--disabled"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches error state snapshot 1`] = `
<div
  class="dataTable dataTable--default dataTable--error"
>
  <div
    class="dataTable__errorContent"
  >
    <div
      class="dataTable__errorTitle"
    >
      Error
    </div>
    <div
      class="dataTable__errorMessage"
    />
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches hover state snapshot 1`] = `
<div
  aria-disabled="false"
  class="dataTable dataTable--default dataTable--sizeDefault"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="dataTable__scrollContainer"
  >
    <table
      class="dataTable__table"
      role="table"
    >
      <thead
        class="dataTable__thead"
      >
        <tr>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="dataTable__tbody"
      >
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches loading state snapshot 1`] = `
<div
  class="dataTable dataTable--loading"
>
  <div
    class="dataTable__loadingContent"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-loader-circle dataTable__spinner"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
    <span
      class="dataTable__loadingText"
    >
      Loading table data...
    </span>
    <span
      class="dataTable__srOnly"
      role="status"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches required state snapshot 1`] = `
<div
  aria-disabled="false"
  class="dataTable dataTable--default dataTable--sizeDefault"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
  required=""
>
  <div
    class="dataTable__scrollContainer"
  >
    <table
      class="dataTable__table"
      role="table"
    >
      <thead
        class="dataTable__thead"
      >
        <tr>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="dataTable__tbody"
      >
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataTable > Snapshots > matches selected state snapshot 1`] = `
<div
  aria-disabled="false"
  class="dataTable dataTable--default dataTable--sizeDefault"
  data-error="false"
  data-loading="false"
  data-testid="datatable"
>
  <div
    class="dataTable__scrollContainer"
  >
    <table
      class="dataTable__table"
      role="table"
    >
      <thead
        class="dataTable__thead"
      >
        <tr>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            class="dataTable__th"
            role="columnheader"
            tabindex="-1"
          >
            <div
              class="dataTable__thContent"
            >
              <span>
                Value
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="dataTable__tbody"
      >
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 1
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            A
          </td>
        </tr>
        <tr
          class="dataTable__row"
          role="row"
        >
          <td
            class="dataTable__td"
            role="cell"
          >
            Test Item 2
          </td>
          <td
            class="dataTable__td"
            role="cell"
          >
            B
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
